### ISSUE TITLE
Feature: Add function-of-interest reachability lookup 

### ISSUE THREAD

**Issue by Navidem** [ossf/fuzz-introspector#18]
It will be useful to employ the reachability data in a way that the user can lookup a function-of-interest to find out which recommended fuzz target may reach to the FOI.

**Comment by DavidKorczynski** [comment]
This is already done? The function overview table has a search functionality so a user can just search for their function name and get a lot of details about it, e.g. the number of fuzzers that reach it.

**Comment by DavidKorczynski** [comment]
In general, the reason for making the tables searchable and sortable was to handle lookup tasks easily with respect to fairly "quantitative data points". This is also convenient for extensions in that we can simply add a new data entry in the table and get search/sort for free

**Comment by Navidem** [comment]
> Am referring to this table specifically: https://github.com/ossf/fuzz-introspector/blob/main/img/functions_overview.png

Yes, this is useful. But my point is for the suggestion section (Target function analysis), where we are proposing potential new fuzz targets. For there we can provide a search feature allowing the user to lookup which one of proposed targets may reach their FOI.

**Comment by DavidKorczynski** [comment]
Ah cool, I understand now!

We already have a table at the bottom of the HTML report that shows data about all functions in the project as it will be if the suggested fuzzers are adopted. This is the code: https://github.com/ossf/fuzz-introspector/blob/main/post-processing/fuzz_html.py#L610-L621

This table shows "fuzzer hit count" for each function - we can add a column that show explicitly the fuzzers that hit a given function. Then, the search feature can be used to match suggested functions with functions of interest?

**Comment by DavidKorczynski** [comment]
Another thought on this is we should watch out not to explode the HTML page. It makes sense to refactor first https://github.com/ossf/fuzz-introspector/issues/3 - in parallel with that we could create an individual page for each suggested fuzzer which can hold a lot of information about the suggested fuzzer,

**Comment by Navidem** [comment]
> This table shows "fuzzer hit count" for each function - we can add a column that show explicitly the fuzzers that hit a given function. Then, the search feature can be used to match suggested functions with functions of interest?

This should work.

**Comment by Navidem** [comment]
The first version of the feature is here 132d2ea623cb3f42f109067582d5e88a423280c3 (not yet merged).
Along with updating hitcount for both actual fuzz targets and the suggested ones, reached_by_fuzzers property is populated for each function.

Once potential problem is where a project have too many fuzz targets and the table entry for a commonly reached function becomes so big (like the attached screenshot for binutils project).

<img width="1402" alt="Screen Shot 2022-02-16 at 10 42 51 AM" src="https://user-images.githubusercontent.com/19780488/154380713-b0ccea89-32ab-44ca-a0eb-2000478a16ed.png">


A solution is having collapsible entry for the specific table entry.

**Comment by DavidKorczynski** [comment]
I would probably suggest a different strategy.

One of the things I refactored recently is trying to set up "analysis interfaces", which are thought to be like analysis plugins. The current plugins are listed here: https://github.com/ossf/fuzz-introspector/blob/93e401ed6d4db13ea7aeca14a35d0e150a334b7b/post-processing/fuzz_html.py#L760-L764

What we could do with the function lookup is to create a separate analysis interface for it. We could then locate the result of this analysis in their own `.html` file similar to how I did with the calltree here: https://github.com/ossf/fuzz-introspector/blob/93e401ed6d4db13ea7aeca14a35d0e150a334b7b/post-processing/fuzz_html.py#L389-L394

You can just avoid adding it to the default list while it is being developed on: https://github.com/ossf/fuzz-introspector/blob/93e401ed6d4db13ea7aeca14a35d0e150a334b7b/post-processing/main.py#L72-L75

In this manner you could just add an arbitrary amount of data in that file and then create a link to this specific page from the main HTML report. 

I think it would be nice to try and phase out some of the data that is on the main report, simply because it will very quickly get bloated -- and we're now getting concerned with space issues. This is kind of a blocker since we're doing a lot of exploratory stuff at the moment, as we're not sure yet what the absolute most useful information is to show the user with just a limited amount of space. 

As such, I would suggest you create a whole new analysis interface as the existing ones. Then, down the line (~six months +/- ?), we can handpick what information is the absolute best to show users without giving them a bloated report. This approach would allow you not to be concerned about spacing etc. right now and also try out some ideas without committing fully to it.

**Comment by Navidem** [comment]
Thanks David for the suggestions.

> One of the things I refactored recently is trying to set up "analysis interfaces", which are thought to be like analysis plugins.

This is a very good change, as you mentioned gives us freedom to develop separate analysis plugins.

> As such, I would suggest you create a whole new analysis interface as the existing ones. Then, down the line (~six months +/- ?), we can handpick what information is the absolute best to show users without giving them a bloated report. This approach would allow you not to be concerned about spacing etc. right now and also try out some ideas without committing fully to it.

This is what I really like and will do with analysis ideas that I have like branch profiling and data flow based drilling.

That said, about the current change that I've made to present list of reaching fuzzers per functions, I believe it belongs to the current `OptimalTargets` analysis. Because it is highly related to what we have now on the portal in terms of project overview, and informing about the fuzz target's reachability.

Thanks to @oliverchang for the pointer, I was able to incorporate collapsible and it made the table very more readable (we may even use it for "Args" list).

<img width="1142" alt="Screen Shot 2022-02-17 at 2 55 53 PM" src="https://user-images.githubusercontent.com/19780488/154588054-360e23ae-c1e8-4313-bc2f-5cfc2e320152.png">
